diesel_codegen/src/util.rs

```diff
@@ -101,22 +101,29 @@ pub fn inner_of_option_ty(ty: &Ty) -> Option<&Ty> {
     }
 }
 
-pub fn get_options_from_input(attrs: &[Attribute], on_bug: fn() -> !)
-    -> Option<&[MetaItem]>
+pub fn get_options_from_input(attrs: &Vec<Attribute>, on_bug: fn() -> !)
+    -> Option<Vec<MetaItem>>
 {
     let options = attrs.iter().find(|a| a.name() == "options").map(|a| &a.value);
     match options {
-        Some(&MetaItem::List(_, ref options)) => Some(options),
+        Some(&MetaItem::List(_, ref options)) => {
+            Some(options.iter().map(|o| {
+                match o {
+                   &NestedMetaItem::MetaItem(ref m) => m.clone(),
+                   _ => on_bug(),
+                }
+            }).collect())
+        }
         Some(_) => on_bug(),
         None => None,
     }
 }
 
-pub fn get_option<'a>(
-    options: &'a [MetaItem],
+pub fn get_option(
+    options: &Vec<MetaItem>,
```

Comment:
I could see the return types needing to change. Unclear on why the input type should have needed to change. There is no case where `&Vec<MetaItem>` wouldn't deref to `&[MetaItem]`